python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -419,10 +418,98 @@ def deduplicates(self):
             data_shards = spark_df_to_pd_sparkxshards(distinctDF)
             return data_shards
         else:
-            # we may support numpy or other types later
             invalidInputError(False,
                               "Currently only support dedup() on XShards of Pandas DataFrame")
 
+    def sort_values(self, col_names=None, ascending=True):
+        """
+        Sort the value of shards. This is only applicable for SparkXShards of Pandas Series.
+        
+        :param col_names list of column or column naames to sort by
+        :param ascending bool, default True. Specify sort orders
+        :return: a new SparkXShards sorted by the specified columns.
+        """
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
+            sort_df = df.sort(col_names, ascending=ascending)
+            sort_df = sort_df.coalesce(1)
+            data_shards = spark_df_to_pd_sparkxshards(sort_df)
+            return data_shards
+        else:
+            invalidInputError(False,
+                              "Currently only support sort() on XShards of Pandas DataFrame")
+
+    def max_values(self, col_name):
+        """
+        Get the max values of the column name. This is only applicable for SparkXShards
+         of Pandas Series.
+
+        :param col_name column name that need return the max value
+        :return: maximum for the specified columns.
+        """
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
```

Comment:
do we need to convert to spark df here? This can be simply implemented as a reduce function on RDD, yes?

Reply:
Yes and have used reduce function to get the max value.